### PR TITLE
Add tests for ignoring properties case for ordering. Related to #78

### PR DESCRIPTION
**Pull request by Astrael1**
Hello :)
There are some issues with #78 
1. Shorthands
I don't know how the plugin should behave with shorthands. I assumed that if user types

`Border-width: 1px; 
Border-top-width: 2px;`

they do it on purpose and they expect it to behave just like any other css properties. Do you agree with that?

2. It needs [postcss-sorting](https://github.com/hudochenkov/postcss-sorting) to have an option to ignore case. Shall I report that issue there?
3. Do you have any more tests ideas? Are you willing to merge the tests without other changes?

### CODE REVIEW

File: rules/properties-alphabetical-order/tests/index.js [hunk 1]
```diff
@@ -68,6 +68,21 @@ testRule({
...
+		},
+		{
+			// blocked by https://github.com/hudochenkov/stylelint-order/issues/78
+			skip: true,
+			code: 'a { display: block; Margin: 0 auto 5px 0; margin: 0 auto 5px 0; width: auto; }',
+		},
+		{
+			// blocked by https://github.com/hudochenkov/stylelint-order/issues/78
+			skip: true,
+			code: 'a { align: center; Border-width: 1px; Border-top-width: 2px; color: red; }',
```
Comment (hudochenkov) [inline rules/properties-alphabetical-order/tests/index.js:84]: I would also add cases:

```
border-width: 1px; Border-top-width: 2px;
Border-width: 1px; border-top-width: 2px;
```

File: rules/properties-alphabetical-order/tests/index.js [hunk 2]
```diff
@@ -131,6 +146,13 @@ testRule({
 			fixed: '@media print { color: red; top: 0; }',
 			message: messages.expected('color', 'top'),
 		},
+		{
```
Comment (hudochenkov) [inline rules/properties-alphabetical-order/tests/index.js:149]: We need more tests for `reject`. You could take the first case and add variation when the first property capitalized, and variation, where the second.

**Comment by hudochenkov** [comment]
1. Shorthands area already supported https://github.com/hudochenkov/stylelint-order/blob/416d00c13c416a3529f712f3dd8cd0fbf63ea909/rules/checkAlphabeticalOrder.js#L11-L19 I don't expect anything new needed for this.
2. Correct. First `postcss-sorting` need to ignore case, then we can add it to `stylelint-order`. Because `postcss-sorting` powering autofixing, it would be strange if logic for checking order in `stylelint-order` would be different than its fixing functionality.

File: rules/properties-alphabetical-order/tests/index.js (hunk 1 above)
Comment (Astrael1) [reply rules/properties-alphabetical-order/tests/index.js:84]: obraz
If above properties are correct, then the shorthand comparison must be altered
https://github.com/hudochenkov/stylelint-order/blob/416d00c13c416a3529f712f3dd8cd0fbf63ea909/rules/checkAlphabeticalOrder.js#L11-L19

Comment (hudochenkov) [reply rules/properties-alphabetical-order/tests/index.js:84]: That's why we need those tests :)

But I think once config values and properties names will be lower-cased before comparison, everything will just work.

File: rules/properties-order/tests/validate-options.js [hunk 3]
```diff
@@ -121,3 +121,33 @@ testConfig({
...
+	// blocked by https://github.com/hudochenkov/stylelint-order/issues/78
+	skip: true,
+	ruleName,
+	description: 'no duplicates in properties (case insensitive)',
+	valid: false,
+	config: [
+		{
+			emptyLineBefore: 'always',
+			order: 'flexible',
+			properties: ['top', 'Top'],
```
Comment (Astrael1) [inline rules/properties-order/tests/validate-options.js:149]: @hudochenkov I need your decision on this. User may provide above properties. In this case they should be informed, that the plugin doesn't work like that.

**Comment by Astrael1** [comment]
I added some tests. I haven't added tests in [grouped-strict](https://github.com/hudochenkov/stylelint-order/blob/master/rules/properties-order/tests/grouped-strict.js) yet. 

I'd like to know if it's fine. Are there too many tests or not enough etc?

File: rules/properties-order/tests/flat.js [hunk 4]
```diff
@@ -199,9 +237,19 @@ testRule({
 		{
 			code: 'a { top: 0; height: 1px; color: pink; }',
 		},
+		{
```
Comment (hudochenkov) [inline rules/properties-order/tests/flat.js:240]: Could remove this one and every other test below, that was added in this file .

#78 is about very minor edge case. We don't need so many tests. Code will be already covered by other tests.

But I appreciate your attention to details :)

File: rules/properties-order/tests/grouped-flexible.js
Comment (hudochenkov) [inline rules/properties-order/tests/grouped-flexible.js:1]: I don't think we need these tests.

File: rules/properties-order/tests/validate-options.js
Comment (hudochenkov) [inline rules/properties-order/tests/validate-options.js:1]: These tests are not needed. As they are out of scope of #78.

**Comment by hudochenkov** [comment]
> I haven't added tests in [grouped-strict](https://github.com/hudochenkov/stylelint-order/blob/master/rules/properties-order/tests/grouped-strict.js) yet.

They are not needed.

**Review by hudochenkov — APPROVED** [review]
Thank you!
